README.md

```diff
@@ -21,90 +21,75 @@ Here are some of the benefits of using this library:
 **API Reference**: https://go.microsoft.com/fwlink/?linkid=2091700
 
 ### Feature Flags
-Feature flags are composed of two parts, a name and a list of feature-filters that are used to turn the feature on.
+Feature flags can either be on or off. They are composed of two parts, a name and a list of feature-filters that are used to turn the feature on.
 
 ### Feature Filters
-Feature filters define a scenario for when a feature should be enabled. When a feature is evaluated for whether it is on or off, its list of feature-filters are traversed until one of the filters decides the feature should be enabled. At this point the feature is considered enabled and traversal through the feature filters stops. If no feature filter indicates that the feature should be enabled, then it will be considered disabled.
+Feature filters define a scenario for when a feature flag should be enabled. When a feature flag is evaluated for whether it is on or off, its list of feature-filters are traversed until one of the filters decides the feature flag should be enabled. At this point the feature flag is considered enabled and traversal through the feature filters stops. If no feature filter indicates that the feature flag should be enabled, then it will be considered disabled.
 
 As an example, a Microsoft Edge browser feature filter could be designed. This feature filter would activate any features it is attached to as long as an HTTP request is coming from Microsoft Edge.
 
+### Dynamic Features
+Dynamic features can have values who's type range from object, to string, to integer and so on. Additionally, dynamic features can have an unlimited amount of values. A developer is free to choose what type should be returned when the value of a dynamic feature is requested. They are also free to choose how many options, known as variants, are available to select from.
+
+### Feature Variants
+Feature variants are the different versions of a feature that could be returned when the value of a dynamic feature is requested. Beyond the value of the feature, a feature variant contains information describing under what circumstances it should be returned over other available variants.
+
+### Feature Variant Assigners
+A feature variant assigner is a component that uses contextual information within an application to decide which feature variant should be chosen when a variant of a dynamic feature is requested.
+
 ## Registration

```

Comment:
The content explained the fundamental design of the feature management library that is built upon IConfiguration. I am not sure why the title is called "Registration". Maybe we can have a better name?

The content applies to both feature flag and dynamic features, but everything nested under this bullet point is about feature flags. We probably should have a new section "Feature flags" and have everything nested under it instead?

Reply:
It does look like it needs a restructure. What's under `Registration` right now goes into details about how IConfiguration is the backbone of the library. I think it can be renamed to `Feature Configuration` and be a bullet in the same vein as "FeatureFlags", Feature Filters, "Dynamic Features" and "FeatureVariants" above it. These sections are all like sneak-peeks with short break downs of the libraries offering.

Then we can have the "Feature Flags" section  below it that is a header 2 `##` instead of `###`. This explains how to define feature flags etc.

Reply:
This sounds great!